plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/AwsClientManager.kt

```diff
@@ -90,8 +90,9 @@ open class AwsClientManager : ToolkitClientManager(), Disposable {
         val userAgent: String by lazy {
             val platformName = tryOrNull { ApplicationNamesInfo.getInstance().fullProductNameWithEdition.replace(' ', '-') }
             val platformVersion = tryOrNull { ApplicationInfoEx.getInstanceEx().fullVersion.replace(' ', '-') }
-            val pluginVersion = tryOrNull { PluginManager.getPluginByClass(this::class.java)?.version }
-            "AWS-Toolkit-For-JetBrains/$pluginVersion $platformName/$platformVersion ClientId/${AwsSettings.getInstance().clientId}"
+            val pluginName = PluginResolver.getInstance().product.toString().replace(" ", "-")
+            val pluginVersion = tryOrNull { PluginResolver.getInstance().version }
+            "$pluginName/$pluginVersion $platformName/$platformVersion ClientId/${AwsSettings.getInstance().clientId}"
```

Comment:
i think the SDKs are mostly independent between extensions where it's not really a big problem, but there's also the case where a given client is used by both extensions so we'd need to have the user agent be dynamic on the caller

Reply:
Good point. Still working out a way to provide a UA at the request level. As of the current change they will be generated per SDK client instance.

Reply:
Addressed this